lib/subcommands/keychain.js

```diff
@@ -1,35 +1,83 @@
+import { fs, tempDir } from 'appium-support';
+import path from 'path';
+import _ from 'lodash';
+
+
 const commands = {};
 
+async function handleRawPayload (payload, onPayloadStored) {
+  const tmpRoot = await tempDir.openDir();
+  try {
+    const filePath = path.resolve(tmpRoot, 'certificate.pem');
+    if (_.isBuffer(payload)) {
+      await fs.writeFile(filePath, payload);
+    } else {
+      await fs.writeFile(filePath, payload, 'utf8');
+    }
+    await onPayloadStored(filePath);
+  } finally {
+    await fs.rimraf(tmpRoot);
+  }
+}
+
+
+/**
+ * @typedef {Object} CertOptions
+ * @property {boolean} raw [false] - whether the `cert` argument
+ * is the path to the certificate on the local file system or
+ * a raw certificate content
+ */
+
 /**
  * Adds the given certificate to the Trusted Root Store on the simulator
  *
  * @since Xcode 11.4 SDK
- * @param {string} certPath the full path to a valid .cert file containing
- * the certificate content
+ * @param {string} cert the full path to a valid .cert file containing
+ * the certificate content or the certificate content itself, depending on
+ * options
+ * @param {CertOptions} opts
  * @throws {Error} if the current SDK version does not support the command
  * or there was an error while adding the certificate
  * @throws {Error} If the `udid` instance property is unset
  */
-commands.addRootCertificate = async function addRootCertificate (certPath) {
-  await this.exec('keychain', {
+commands.addRootCertificate = async function addRootCertificate (cert, opts = {}) {
+  const {
+    raw = false,
+  } = opts;
+  const execMethod = async (certPath) => await this.exec('keychain', {
     args: [this.requireUdid('keychain add-root-cert'), 'add-root-cert', certPath],
   });
+  if (raw) {
+    await handleRawPayload(cert, execMethod);
+  } else {
+    await execMethod(cert);
+  }
 };
 
 /**
  * Adds the given certificate to the Keychain Store on the simulator
  *
  * @since Xcode 11.4 SDK
- * @param {string} certPath the full path to a valid .cert file containing
- * the certificate content
+ * @param {string} cert the full path to a valid .cert file containing
+ * the certificate content or the certificate content itself, depending on
+ * options
+ * @param {CertOptions} opts
  * @throws {Error} if the current SDK version does not support the command
  * or there was an error while adding the certificate
  * @throws {Error} If the `udid` instance property is unset
  */
-commands.addCertificate = async function addCertificate (certPath) {
-  await this.exec('keychain', {
+commands.addCertificate = async function addCertificate (cert, opts = {}) {
+  const {
+    raw = false,
+  } = opts;
+  const execMethod = async (certPath) => await this.exec('keychain', {
     args: [this.requireUdid('keychain add-cert'), 'add-cert', certPath],
   });
+  if (raw) {
+    await handleRawPayload(cert, execMethod);
```

Comment:
Beat me to answering your question. :)

But also, is there a reason that the `execMethod` function needs to be called as a callback? Does it not work to do something like
```js
let certPath = cert;
if (raw) {
  certPath = await handleRawPayload(cert);
}
await execMethod(certPath);
```

Reply:
no, because I would like to make sure the temporary folder is deleted after the whole block is executed

Reply:
Makes sense. Thanks!